tests/e2e/specs/edit-posts.test.js

```diff
@@ -53,24 +49,24 @@ test.describe( 'Edit Posts', () => {
 		await editor.publishPost();
 		await admin.visitAdminPage( '/edit.php' );
 
-		await page.waitForSelector( '#the-list .type-post' );
+		const listTable = page.getByRole( 'table', { name: 'Table ordered by' } );
+		await expect( listTable ).toBeVisible();
 
 		// Click the post title (edit) link
-		await page.locator( `//a[contains(@class, "row-title")][contains(text(), "${ title }")]` ).click();
+		await listTable.getByRole( 'link', { name: `“${ title }” (Edit)` } ).click();
 
 		// Wait for the editor iframe to load, and switch to it as the active content frame.
-		const editorFrame = await page.waitForSelector( 'iframe[name="editor-canvas"]' );
-
-		const innerFrame = await editorFrame.contentFrame();
+		await page
+				.frameLocator( '[name=editor-canvas]' )
+				.locator( 'body > *' )
+				.first()
+				.waitFor();
```

Comment:
I know this is how we do it in Gutenberg, but I wonder if going down the extra `> *` level is necessary. Shouldn't we achieve the same result with the following?

```js
await page.frameLocator('[name=editor-canvas]').locator('body').waitFor();
```

(This is a workaround because the `frameLocator` API doesn't implement a `waitFor` methor.)


cc: @kevin940726

Reply:
I don't have a strong opinion on this one. I copied it from the createNewPost helper.

Reply:
I recall in some weird cases the canvas renders an empty `<body>` but then remounts itself and renders a different document. I believe it's a workaround of how the editor frame works now. I don't know if the issue has been resolved though.

Reply:
Interesting, I'd assume that since it's lazily evaluated that wouldn't make a difference. Any idea where that issue occurred?

Reply:
I'm guessing it's due to some internals of block editors `IFrame` component.